### PR TITLE
Fix libc_optz in EMCC_FORCE_STDLIBS builds

### PR DESCRIPTION
**Pull request by kripken**
Aside from `MAIN_MODULE=1`, if the user uses `EMCC_FORCE_STDLIBS` then
all system libraries will be linked in, and potentially cause problems.
Fixes `corez.test_dylink_syslibs_all` as well as `ltoz` and `thinltoz`, see
https://github.com/emscripten-core/emscripten/pull/17179#issuecomment-1158226153

Second part of the diff moves the disabling check into the library, to avoid redundancy
(which would require fixing this in 2 places).

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -1861,13 +1867,7 @@ def add_library(libname):
...
   if settings.PRINTF_LONG_DOUBLE:
     add_library('libprintf_long_double')
-  # Becuase libc_optz overrides parts of libc, it is not compatible with `LINKABLE`
-  # (used in `MAIN_MODULE=1`) because with that setting we use `--whole-archive` to
-  # include all system libraries.
-  # However, because libc_optz is a size optimization it is not really important
-  # when used with `MAIN_MODULE=1` (which links in all system libraries, leading
-  # to overheads far bigger than any savings from libc_optz)
-  if settings.SHRINK_LEVEL >= 2 and not settings.LINKABLE:
+  if settings.SHRINK_LEVEL >= 2:
```
Comment (sbc100) [inline tools/system_libs.py:1870]: I thought you would need to replicate the same logic here as in `can_use()`...  but maybe it never gets to this line in some cases.

**Review by sbc100 — COMMENTED** [review]
lgtm if the tests pass

**Review by sbc100 — APPROVED** [review]
Can you add those failing the tests to the github CI to prove this fix works?

File: tools/system_libs.py (hunk 1 above)
Comment (kripken) [reply tools/system_libs.py]: Yeah, I was wrong... We do need the duplication. Added .

**Comment by kripken** [comment]
Test added.

**Pull request merged**